### ISSUE TITLE
Performance issue with CacheClient initialization on AWS Lambda (Node.js v18)

### ISSUE THREAD

**Issue by kiennt91** [momentohq/client-sdk-javascript#1289]
## Environment
- **AWS Lambda**
- **Node.js version**: `18`
- **@gomomento/sdk version**: `1.86.0`
- **Memory allocated to Lambda**: `512MB`

## Description:
I am encountering a significant delay when initializing the `CacheClient` from `@gomomento/sdk`. The initialization process is taking approximately 10 seconds, which seems unusually long for AWS Lambda. Below you can find the memory statistics before and after the `CacheClient` initialization:

### Memory Usage Before Initialization:
```json
{
    "rss": "120.44 MB",
    "heapTotal": "57.61 MB",
    "heapUsed": "42.83 MB",
    "external": "2.61 MB",
    "arrayBuffers": "0.19 MB"
}
```

### Memory Usage After Initialization:
```json
{
    "rss": "262.25 MB",
    "heapTotal": "182.63 MB",
    "heapUsed": "133.74 MB",
    "external": "2.49 MB",
    "arrayBuffers": "0.07 MB"
}
```

## Code Snippet:
Below is the code snippet used for initializing `CacheClient`:

```javascript
Logger.debug('Initializing MomentoCacheClient...', this.props, Logger.jsonMask(['momentoApiKey']));
this.cacheClient = await CacheClient.create({
  configuration: Configurations.Laptop.v1(),
  credentialProvider: CredentialProvider.fromString({
    apiKey: this.props.momentoApiKey,
  }),
  defaultTtlSeconds: this.props.defaultTtlSeconds,
});
Logger.debug('MomentoCacheClient initialized.', '');
```

## Additional Information:
Here is the log image that shows the delay in initialization: 
![331770767-e8b3a206-7563-4a67-bc8f-f2efa9c1dacd (1)](https://github.com/momentohq/client-sdk-javascript/assets/5936841/add5ed9b-c4c5-4d2a-950c-9e0d6e52e43d)

## Concern:
Is anyone else experiencing similar issues with `@gomomento/sdk`? Any help or insight into the cause of this problem would be greatly appreciated.

**Comment by cprice404** [comment]
Hi @kiennt91 - this is definitely not expected. We are happy to take a look into it.

Can you provide some additional information about how you are packaging your lambda?

**Comment by bruuuuuuuce** [comment]
Hi @kiennt91  - I have tried to reproduce this both locally and inside of a lambda environment without any luck. I was seeing `create` times at ~200ms. The slowest I saw was 700ms but that was in a 128mb lambda. 

A few things to note/I would try

- if you enable debug/trace logging on the client, you might be able to see exactly where its hanging
```javascript
Logger.debug('Initializing MomentoCacheClient...', this.props, Logger.jsonMask(['momentoApiKey']));
this.cacheClient = await CacheClient.create({
  configuration: Configurations.Laptop.v1(
      new DefaultMomentoLoggerFactory(DefaultMomentoLoggerLevel.TRACE)
  ),
  credentialProvider: CredentialProvider.fromString({
    apiKey: this.props.momentoApiKey,
  }),
  defaultTtlSeconds: this.props.defaultTtlSeconds,
});
Logger.debug('MomentoCacheClient initialized.', '');
```
We recently released some additional trace logging for grpc connection state in version [1.88.0](https://github.com/momentohq/client-sdk-javascript/releases/tag/v1.88.0), it is probably worth updating your momento dependency first before enabling trace logging

- how are you bundling your lambda? There is a blog post we wrote [here](https://www.gomomento.com/blog/how-we-turned-up-the-heat-on-node-js-lambda-cold-starts) about how to use `esbuild` to minimize package size and overall speed up nodejs lambdas, and a corresponding [git repo](https://github.com/bruuuuuuuce/lambda-template) with examples on bundling nodejs lambas
- I see you are in the `ap-northeast-1` aws region, are you trying to create caches in this same region? Turning on the `trace` logs should help verify if you are or not
- We have special configuration objects build for a lambda environment, it might be worth trying this and seeing if performance has improved
```javascript
this.cacheClient = await CacheClient.create({
  configuration: Configurations.Lamba.latest(
      new DefaultMomentoLoggerFactory(DefaultMomentoLoggerLevel.TRACE)
  ),
  credentialProvider: CredentialProvider.fromString({
    apiKey: this.props.momentoApiKey,
  }),
  defaultTtlSeconds: this.props.defaultTtlSeconds,
});
```

Let me know if this helps out at all, or if you are still facing any performance issues
